chapters/synchronous.tex

```diff
@@ -424,6 +425,18 @@ \section{Clock Constructors}\label{clock-constructors}
 \end{lstlisting}
 \end{example}
 
+\begin{example}
+Demonstrating the subtle effect of the additional \lstinline!pre!:
+\begin{lstlisting}[language=modelica]
+  Boolean b = time >= 0.5;
+  Clock c = Clock(b);
+  Boolean b2 = sample(b, c);
+\end{lstlisting}
+The first tick for \lstinline!b2! is at \lstinline!0.5! where it is \lstinline!true!.
+This is because both the event clock and sample introduce a delay of one event iteration, keeping them synchronized.
+\end{example}
+
```

Comment:
Can you walk me through this example? It is so subtle. It says "the first tick for b2..." Does b2 also tick? I thought tick would only apply to clock variables. Based on the added text above I am expecting to see 2 events. But I am only seeing an event at 0.5. Events are not my specialty. Or does the text on line 418 mean that there is iteration during an event but the time is stopped?

Reply:
> Can you walk me through this example? It is so subtle. It says "the first tick for b2..." Does b2 also tick? I thought tick would only apply to clock variables. 

Yes, should write something like:

The first tick of c (the clock of b2) is at 0.5 where b2 is true.

> Based on the added text above I am expecting to see 2 events. But I am only seeing an event at 0.5. Events are not my specialty. Or does the text on line 418 mean that there is iteration during an event but the time is stopped?

The latter.

Reply:
Note that it subtly said "the first tick for b2", not "the first tick of b2". However, that was so subtle that I missed it myself.

Reply:
Thanks,  I came to the understanding by the end of my comment, but it's good to confirm it. Resolving.